include/aos/common/tools/array.hpp

```diff
@@ -379,7 +379,7 @@ class Array {
             return ErrorEnum::eInvalidArgument;
         }
 
-        for (auto i = 0; i < end() - pos; i++) {
+        for (auto i = end() - pos - 1; i >= 0; i--) {
```

Comment:
Why it doesn't work in normal order?

Reply:
In the normal order, the elements at the beginnig override elements in the middle of the range.
eg {0 1 2  3} , insert 4 at pos 0 => result = {0 0 0 0}